### ISSUE TITLE
A11y: Exit focused navigation with escape for keyboard users

### ISSUE THREAD

**Issue by RyKilleen** [forem/forem#1154]
**Is your feature request related to a problem? Please describe.**
When a user's context is focused in navigation, modal, or any other pop up menu, it's useful to refocus outside of that context on an Escape key press.

**Additional context**

**Comment by Link2Twenty** [comment]
Here are the aria-practices pages for modal, menu and menu buttons.
https://www.w3.org/TR/wai-aria-practices/#dialog_modal
https://www.w3.org/TR/wai-aria-practices/#menu
https://www.w3.org/TR/wai-aria-practices/#menubutton

**Comment by jessleenyc** [comment]
Thanks for flagging! This issue is approved for anyone who might want to work on it.

**Comment by Zhao-Andy** [comment]
Not stale, still valid.

**Comment by maestromac** [comment]
Sorry again, not stale.

**Comment by Link2Twenty** [comment]
Not stale

**Comment by RyKilleen** [comment]
I'll attempt to tackle this over break, before the new year!

**Comment by nickytonline** [comment]
@RyKilleen thanks so much for the issue. Since Hacktoberfest starts tomorrow, if you're up for taking this on just say the word.

**Comment by RyKilleen** [comment]
I'm spread thin across a few things atm and don't want to monopolize it!

If anyone else would like to give it a go, I'll defer to them. I'll also send it around

**Comment by reobin** [comment]
#10468 adds a hook for easy global key events. If it is to be merged, I'd be up to use it on this issue

**Comment by marcysutton** [comment]
I can take this one on, since I was about to file an issue for it!

**Comment by nickytonline** [comment]
Sounds great @marcysutton. Assigning this one to you.

**Comment by cmgorton** [comment]
@forem/frontend Another one we may want to consider for internal review. Let me know what you think with this issue? It may be close to finished since I know Marcy was working on it before and has a PR for it that needs some changes.

**Comment by nickytonline** [comment]
I think this is a great candidate @cmgorton

**Comment by aitchiss** [comment]
This ties in to #12297 (Optimise all uses of crayons dropdown for accessibility) as well as the open PR from Marcy - #11509

**Comment by mstruve** [comment]
@aitchiss will this be solved by the Marcy PR you mentioned or will their still be more work to do beyond that?

**Comment by aitchiss** [comment]
@mstruve this sprawls a few other open issues, and Marcy's PR (#12297) only addresses one part of it (the heading navigation dropdown). Beyond that, we'd still need to address other dropdowns in the app and usage of modals too.

There's a few other open issues for those things:
- Fixing dropdowns elsewhere in the app: #12297 (Optimise all usage of crayons dropdown for accessibility)
- And modals: #10610 (a11y: improve keyboard access on modals) & #12427 (Forem editor modals are not accessible)

**Comment by mstruve** [comment]
@aitchiss would #12297 and #10610 cover everything from this issue? It feels like it might be a duplicate for #12297 and that is a bit more detailed and thorough in the approach which would lean me towards closing this one in favor of the new one. Thoughts?

**Comment by RyKilleen** [comment]
You should probably close this! It was opened pre-Forem and it's a pretty broad comment, made before there were many folks focusing on this work.

**Comment by aitchiss** [comment]
@mstruve just noticed a commit comment from Marcy's PR has automatically closed this, and I think you're right, we can leave it closed as the remaining work is detailed on the other issues mentioned 👍
